ggml-cuda.cu

```diff
@@ -4619,23 +4619,25 @@ static __global__ void rope_neox(
     }
 
     const int row = blockDim.x*blockIdx.x + threadIdx.x;
-    const int i = row*ncols + col/2;
+    const int ib = col / n_dims;
+    const int ic = col % n_dims;
+
+    const int i = row*ncols + ib*n_dims + ic/2;
     const int i2 = row/p_delta_rows;
 
-    // simplified from `(ib * ncols + col) * (-1 / ncols)`, where ib is assumed to be zero
-    const float cur_rot = -float(col)/ncols;
+    float cur_rot = inv_ndims * ic - ib;
```

Comment:
I think this should be `ib * n_dims + ic`, see #4093

Reply:
I tested the ppl of both versions with falcon 7b and stablelm 3b, and in all cases the result was identical. Maybe they are equivalent?

Reply:
I don't think we need to deal with this in this PR, it wouldn't make sense to change it only in the CUDA implementation. So let's figure a way to test this and solve it in your PR.

Reply:
It only matters when YaRN is enabled. When it isn't, cur_rot is not used. I'm okay with changing it in a later PR.